lib/backend/k8s/resources/customresource.go

```diff
@@ -79,8 +81,17 @@ func (c *customK8sResourceClient) Create(ctx context.Context, kvp *model.KVPair)
 		return nil, K8sErrorToCalico(err, kvp.Key)
 	}
 
+	resOut.GetObjectKind().SetGroupVersionKind(c.k8sResourceTypeMeta.GetObjectKind().GroupVersionKind())
+
+	// Update the return data with the metadata populated by the (Kubernetes) datastore.
+	kvp, err = c.convertResourceToKVPair(resOut)
```

Comment:
Do we need this convertResourceToKVPair function - can't we just call directly through to the resource.go functions to convert the resource?  Seems like there's some surplus code here or further down the line.

Reply:
or if you are gonna keep it then you should include the other bits above and below in the function.

Reply:
We could put this function inline, but since it is called multiple times throughout the other methods, I thought it would be cleaner to keep it all together. There's no real reason that this specifically exists other than to prevent lines of code from being rewritten.

Reply:
Yup - that makes sense.  I was more worried that we seem to have a bit more redirection than I think we need - that was all.  Because the convertResource then calls out to another convert method to do the heavy lifting